20_2 TreeList Row Editing: rewrite (#711)
DevExpress/devextreme-demos: JSDemos/Demos/TreeList/RowEditing/description.md

> The TreeList allows users to edit data in multiple modes. This demo shows the *"row"* edit mode. To edit any row, click Edit in it. Only one row can be in the edit state at a time.

Reviewer: `click Edit in it` Does this refer to an Edit button? 'in it' is also unclear. 

```suggestion
The TreeList allows users to edit data in multiple modes. This demo shows the *"row"* edit mode that allows you to click a row's Edit button to edit the row. Only one row can be in the edit state at a time.
```

Author: The main purpose of this demo is to allow users to edit rows, not click an Edit button to edit rows. That's why can this be:
```suggestion
The TreeList allows users to edit data in multiple modes. This demo shows the *"row"* edit mode that allows you to edit rows. To do this, click a row's Edit button; only one row can be in the edit state at a time.
```
?

Reviewer: I thought the purpose of this demo is to show one of the edit modes? What makes this mode diiferent? Does it enable row editing? Does it add an Edit button to rows?

The TreeList supports different data edit modes. In this demo, *"Row"* edit mode adds an Edit button to each row. Note that only one row can be in the edit state at a time.

To enable row edit mode, set the [mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) property to *"row"* and enable the [editing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) object's [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties to allow edit operations.

> Buttons for allowed edit operations are displayed in an edit column. To customize it programmatically, declare a column of the `buttons` [type](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#type) and specify available buttons. In this demo, the `buttons` column is used to hide an Add button from each row. See the following help topic for more details on edit column customization: [Customize the Edit Column](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Types/Command_Columns/#Customize_the_Edit_Column).

Reviewer 2: The sentence
> To customize it programmatically, declare a column of the `buttons` [type](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#type) and specify available buttons.

Refers to an edit column mentnioned in the previous sentence, correct? If so, should I create another column of the `buttons` type or redefine the mentioned edit column?

Author: This sentence refers to the edit column mentioned in the previous sentence, but this column is not in the columns array because declaring editing.allow... properties creates the edit column _implicitly_. To customize this column, a programmer should declare a column of the `buttons` type add add desired properties. Thus there can be only one column of the buttons type.

Reviewer 2: Alright, let's keep the current version.

Reviewer: ```suggestion
Buttons for allowed edit operations are displayed in an edit column. To customize it programmatically, declare a column of the `buttons` [type](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#type) and specify the available buttons. In this demo, the `buttons` column is used to hide the Add button in each row. See the following help topic for more details on edit column customization: [Customize the Edit Column](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Types/Command_Columns/#Customize_the_Edit_Column).
```

This demo also illustrates the following event handlers:

- [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow)        

> Populates cells of a new row with default values. In this demo, **onInitNewRow** sets John Heart as Head for new rows.

Reviewer: `sets John Heart as Head for new rows` Explain

Author: When the button to add a new row is clicked, the new row editor's Head field is prepopulated:
![image](https://user-images.githubusercontent.com/66619750/114580757-d00d9080-9c87-11eb-84ca-dbd4009d0f1c.png)

Reviewer: ```suggestion
Populates cells of a new row with default values. In this demo, **onInitNewRow** sets John Heart as the initial Head for new rows.
```

- [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing)         
Customizes cell editors. In this demo, this handler disallows users to edit the Head value of John Heart.

> Refer to the following help topic to see the full list of the events raised by edit operations: [Editing Events](/Documentation/Guide/UI_Components/TreeList/Editing/#Events).

Reviewer 2: I believe we can avoid passive voice here. For example:
> Refer to the following topic to see the full list of events that edit operations raise: [Editing Events](/Documentation/Guide/UI_Components/TreeList/Editing/#Events).

To enable the row editing mode, set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) property to *"row"*.

> All edit operations are deactivated in the default configuration. To activate them, set the [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties of the editing object to **true**. This will create the edit column with buttons for each allowed operation on each row. In this demo, we declared a column of type *"buttons"* to add the edit and delete buttons only; the universal button to add a new row is situated in the upper right corner of the demo container. See the [Customize the Edit Column](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Types/Command_Columns/#Customize_the_Edit_Column) article for details.

Reviewer 3: 1. >This will create the edit column with buttons for each allowed operation on each row. 

Is this info really important for developers? I believe they can see the effect in the demo. The same applies to the following sentences: '
In this demo, we declared a column of type *"buttons"* to add the edit and delete buttons only; the universal button to add a new row is situated in the upper right corner of the demo container.'

2. >See the [Customize the Edit Column]() article for details.

It took some time to find some kind of a guideline that tells why this sentence should be rewritten. Please see the screenshot in the next comment.

3. >In this demo, we declared a column of type

Vladimir recommends to avoid first person speech when we can use the imperative mood. I think that we can try to do this as shown below.

```suggestion
All edit operations are deactivated in the default configuration. To activate them, set the [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties of the **editing** object to **true**.

To customize the edit column, assign *"buttons"* to the **column**.[type](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#type) property and specify the button names as shown in this demo. See the following help topic for more details on the edit column customization: [Customize the Edit Column](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Types/Command_Columns/#Customize_the_Edit_Column).
```

Reviewer 3: ![image](https://user-images.githubusercontent.com/33040934/113579842-85f43180-962d-11eb-924c-fb46600efb9c.png)

> Each edit operation raises events. In this demo, we use the [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler to configure the creation of a new row: the Head of a new employee is prepopulated with the company's CEO. We also use the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing) handler to customize the edit form: if a user edits the row with the CEO, the TreeList will disable the Head column editor.

Reviewer 3: 1. >We also use the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing) handler to customize the edit form

It seems to me that we should list the existing events and their purpose and only then show that these events are used for. Moreover, there's no form in this editing mode.

2. We should add a link to the documentation section that lists all events raised by editing.

```suggestion
Each edit operation raises events. The [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler populates an added row with data. In this demo, this handler assigns the CEO's `ID` to the `Head` field of a newly created row. Use the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing) handler to customize a cell's editor. In this demo, we use this handler to disable an editor for the `Head` field when users edit the row with the CEO.

Refer to the following help topic to see the full list of the events raised by editing: [Editing Events](/Documentation/Guide/UI_Components/TreeList/Editing/#Events).
```

To enable the row editing mode, set the **editing**.[mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) property to *"row"*.

> All edit operations are deactivated in the default configuration. To activate them, set the [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties of the **editing** object to **true**. This will create the edit column with buttons for each allowed operation.

Reviewer 4: ```suggestion
The TreeList allows users to edit data in multiple modes. This demo shows the *"row"* edit mode. To start editing any row, click Edit in it. Only one row can be in the edit state at a time.

To enable row edit mode, set the [mode](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) property to *"row"* and enable the [editing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode) object's [allowUpdating](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating), [allowAdding](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowAdding), and [allowDeleting](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowDeleting) properties to allow edit operations.
```

> To customize the edit column, declare a column of the buttons type and specify the button names as shown in this demo. See the following help topic for more details on the edit column customization: [Customize the Edit Column](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Types/Command_Columns/#Customize_the_Edit_Column).

Reviewer 4: ```suggestion
Buttons that start the edit operations are displayed in an edit column. To customize it programmatically, declare a column of the `buttons` [type](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#type) and specify available buttons. In this demo, the `buttons` column is used to hide an Add button from each row. See the following help topic for more details on edit column customization: [Customize the Edit Column](/Documentation/Guide/UI_Components/TreeList/Columns/Column_Types/Command_Columns/#Customize_the_Edit_Column).
```

> Each edit operation raises events. The [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow) handler populates an added row with data. In this demo, this handler assigns the CEO's `ID` to the `Head` field of a newly created row. Use the [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing) handler to customize a cell's editor. In this demo, we use this handler to disable an editor for the `Head` field when users edit the row with the CEO.

Reviewer 4: ```suggestion
This demo also illustrates the following event handlers:

- [onInitNewRow](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onInitNewRow)        
Populates cells of a new row with default values. In this demo, **onInitNewRow** sets John Heart as Head for new rows.

- [onEditorPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onEditorPreparing)         
Customizes cell editors. In this demo, this handler disallows users to edit the Head value of John Heart.
```

> Refer to the following help topic to see the full list of the events raised by editing: [Editing Events](/Documentation/Guide/UI_Components/TreeList/Editing/#Events).

Reviewer 4: -ing
